week1/Q2.py

```diff
@@ -0,0 +1,51 @@
+class Node:
+	def __init__(self, value = None, next = None):
+		self.value = value
+		self.next = next
+
+
+class LinkedList:
+	def __init__(self):
+		self.first = None
+		self.last = None
+		self.length = 0
+
+	def clear(self):
+		self.__init__()
```

Comment:
I think it is generally beset not to call `__init__()` directly, since it is a special method in Python.
Instead, you can call `clear()` from `__init__()` instead, if you prefer.